.github/workflows/code-formatting.yml

```diff
@@ -74,3 +74,24 @@ jobs:
       - name: mypy check
         run: |
           mypy
+
+  python-vulture:
```

Comment:
I've been following the issue for which this PR has been raised and I wonder if it makes sense to have this as a separate workflow instead of adding it in the code formatting workflow.